src/main/java/ca/bc/gov/api/oracle/legacy/controller/ClientController.java

```diff
@@ -0,0 +1,73 @@
+package ca.bc.gov.api.oracle.legacy.controller;
+
+import ca.bc.gov.api.oracle.legacy.dto.ClientPublicViewDto;
+import ca.bc.gov.api.oracle.legacy.service.ClientService;
+import io.swagger.v3.oas.annotations.Parameter;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@CrossOrigin(maxAge = 3600)
+@RestController
+@RequestMapping("/api/clients")
+public class ClientController {
+
+  @Autowired
+  private ClientService clientService;
+
+  @GetMapping(value = "/findByClientNumber/{clientNumber}")
+  public ResponseEntity<ClientPublicViewDto> findByClientNumber(
+      @PathVariable String clientNumber) {
+    return ResponseEntity.ok(clientService.findByClientNumber(clientNumber));
+  }
+
+  @GetMapping("/findAllNonIndividuals")
+  public ResponseEntity<List<ClientPublicViewDto>> findAllNonIndividuals(
+      @RequestParam(defaultValue = "1") int currentPage,
+      @RequestParam(defaultValue = "10") int itemsPerPage,
+      @RequestParam(defaultValue = "clientName") String sortedColumnName) {
+
+    return ResponseEntity.ok(
+        clientService.findAllNonIndividualClients(currentPage, itemsPerPage, sortedColumnName));
+  }
+
+  @GetMapping(value = "/search")
```

Comment:
maybe we should call it "/findByNames"? I feel call it "/search" is too general, I feel search means could search based on anything

Reply:
Yes, I agree with Catherine!

Reply:
We will change the name of the endpoint in the next iteration